checker/src/org/checkerframework/checker/index/samelen/SameLenAnnotatedTypeFactory.java

```diff
@@ -73,6 +77,42 @@ public QualifierHierarchy createQualifierHierarchy(MultiGraphFactory factory) {
         return new SameLenQualifierHierarchy(factory);
     }
 
+    @Override
+    public AnnotatedTypeMirror getAnnotatedTypeLhs(Tree tree) {
+        AnnotatedTypeMirror atm = super.getAnnotatedTypeLhs(tree);
+        if (tree.getKind() == Tree.Kind.VARIABLE) {
+            String varName = null;
+            Element elt = TreeUtils.elementFromDeclaration((VariableTree) tree);
+            if (elt.getKind() == ElementKind.LOCAL_VARIABLE
```

Comment:
Could you please put this code in `internalReprOfVariable` so that is can handle all kinds of VariableTrees?